src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestUpdater.cs

```diff
@@ -1017,21 +1019,26 @@ private async Task ProcessCodeFlowUpdateAsync(
 
         NativePath localRepoPath;
         CodeFlowResult codeFlowRes;
-        string? previousSourceSha;
+        string? previousSourceSha; // is null in some edge cases like onboarding a new repository
 
         try
         {
             if (isForwardFlow)
             {
                 codeFlowRes = await _vmrForwardFlower.FlowForwardAsync(subscription, build, prHeadBranch, cancellationToken: default);
                 localRepoPath = _vmrInfo.VmrPath;
-                previousSourceSha = codeFlowRes.PreviouslyFlownSha;
+                previousSourceSha = (await remote.GetSourceManifestAsync(
+                    subscription.TargetRepository,
+                    subscription.TargetBranch))
+                    .GetRepoVersion(subscription.TargetDirectory)?.CommitSha;  // todo is there a general util method for getting 'mapping name' from repo URL?
```

Comment:
nit: This is a bit hard to read and also where exceptions from this block come will be hard to identify - can we have one line for reading the manifest and another for getting the repo?